keel-titus-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/titus/cluster/TitusClusterHandler.kt

```diff
@@ -208,17 +259,94 @@ class TitusClusterHandler(
         ResourceDiff(desired, current?.get(region))
       }
 
+  private fun TitusClusterSpec.generateOverrides(serverGroups: Map<String, TitusServerGroup>) =
+    serverGroups.forEach { (region, serverGroup) ->
+      val workingSpec = serverGroup.exportSpec()
+      val diff = ResourceDiff(workingSpec, defaults)
+      if (diff.hasChanges()) {
+        (overrides as MutableMap)[region] = TitusServerGroupSpec(
```

Comment:
I wish kotlin had `**kwargs`... 😃

Reply:
Since everything on `TitusServerGroupSpec` has a default value of null, you could do something like:
```kotlin
var newSpec = TitusServerGroupSpec()
if ("capacity" in diff.affectedRootPropertyNames) {
  newSpec = newSpec.copy(capacity = workingSpec.capacity)
}
// continued for each property
```
That would remove all the ` else { null }` bits which take up a lot of space, and I think simplify how this reads.

Reply:
@emjburns great suggestion! will make the fix.